development-guide.md

```diff
@@ -6,30 +6,15 @@ The canisters in this repository are deployed in production by submitting propos
 
 Below are the steps needed to cut a release:
 
-### Bitcoin Canister
-
 1. Locally, from the same commit as the release is intended, build the canisters by running `docker build -t canisters .`.
 2. Extract the wasm of the Bitcoin Canister by running `docker run --rm --entrypoint cat canisters /ic-btc-canister.wasm.gz > ic-btc-canister.wasm.gz`.
-3. Compute the checksum of the canister by running `sha256sum ic-btc-canister.wasm.gz`.
-4. On the [releases page](https://github.com/dfinity/bitcoin-canister/releases), click on "Draft a new release". Make sure the right commit is selected.
-5. Create a new tag with the name: `release/<yyyy-mm-dd>`.
-6. Set the title to be: `Bitcoin Canister Release (<yyyy-mm-dd>)`.
-7. Add release notes. Github can generate the release notes by clicking on "Generated Release Notes". Modify as needed.
-8. Add the commit hash of the Bitcoin Canister's wasm from step 3 to the release notes.
-9. Attach the Bitcoin Canister's wasm to the release notes (and nothing else).
-10. Check the "Set as a pre-release" box to indicate that this release hasn't been deployed to production yet.
-11. Once the Bitcoin mainnet canister is upgraded to this version, uncheck the "Set as a pre-release" box to indicate that it's deployed.
-
-### Watchdog Canister
-
-1. Locally, from the same commit as the release is intended, build the canisters by running `docker build -t canisters .`.
-2. Extract the wasm of the Watchdog Canister by running `docker run --rm --entrypoint cat canisters /watchdog-canister.wasm.gz > watchdog-canister.wasm.gz`.
-3. Compute the checksum of the canister by running `sha256sum watchdog-canister.wasm.gz`.
-4. On the [releases page](https://github.com/dfinity/bitcoin-canister/releases), click on "Draft a new release". Make sure the right commit is selected.
-5. Create a new tag with the name: `watchdog/release/<yyyy-mm-dd>`.
-6. Set the title to be: `Watchdog Canister Release (<yyyy-mm-dd>)`.
-7. Add release notes. Github can generate the release notes by clicking on "Generated Release Notes". Modify as needed.
-8. Add the commit hash of the Watchdog Canister's wasm from step 3 to the release notes.
-9. Attach the Watchdog Canister's wasm to the release notes (and nothing else).
-10. Check the "Set as a pre-release" box to indicate that this release hasn't been deployed to production yet.
-11. Once the Watchdog for the Bitcoin mainnet canister is upgraded to this version, uncheck the "Set as a pre-release" box to indicate that it's deployed.
+3. Extract the wasm of the Watchdog Canister by running `docker run --rm --entrypoint cat canisters /watchdog-canister.wasm.gz > watchdog-canister.wasm.gz`.
```

Comment:
if you don't mind I'd like to change this description, so it has two parts high- and low-level steps, and minimise the amount of steps. also since the total deployment may take some time I think it can help to have the deployment status in release notes with the links to corresponding NNS proposal or a pointer to the previous release. here is my take, feel free to modify it as you like.

```md
# Development Guide

## Release Preparation

The canisters in this repository are deployed in production by submitting proposals to the Internet Computer's [Network Nervous System](https://internetcomputer.org/nns).

Due to limitations in GitHub's release handling, we cannot create separate latest releases for the Bitcoin canister and the Watchdog canister. Therefore, we include both artifacts in each latest release. 

To facilitate clarity, we explicitly specify which of them have changed, or if both have been modified, and provide their release status, accounting for potential deployment durations, such as
 - `no change since release/<yyyy-mm-dd>`
 - `waiting for deployment`
 - `deployed, <NNS proposal link>`

Only after all the expected canisters were deployed the `pre-release` can be turned into a proper `latest release`.

## Steps to Cut a Release

1. Identify the commit for the release, eg. `aff3eef`
2. Draft a new pre-release
    - Click on `Draft a new release` at the [releases page](https://github.com/dfinity/bitcoin-canister/releases), make sure the right commit is selected
    - Create a new tag with the name `release/<yyyy-mm-dd>`
    - Set the title to be `release/<yyyy-mm-dd>`
    - Check the `Set as a pre-release` box to indicate that this release has not been deployed to production yet
    - Add release notes. Github can generate the release notes by clicking on `Generated Release Notes`, modify as needed
3. Prepare canister WASM files and compute their checksums
    - **Note**: there is no guarantee on Mac M1 for reproducible build, preferably use Ubuntu
    ```shell
    # Checkout the repo with a given commit.
    $ git clone https://github.com/dfinity/bitcoin-canister &&\
        cd bitcoin-canister &&\
        git checkout aff3eef

    # Use docker to reproducibly build canister WASMs.
    $ docker build -t canisters .

    # Extract WASM files.
    $ docker run --rm --entrypoint cat canisters /ic-btc-canister.wasm.gz > ic-btc-canister.wasm.gz
    $ docker run --rm --entrypoint cat canisters /watchdog-canister.wasm.gz > watchdog-canister.wasm.gz

    # Verify that the files are present.
    $ ls *.wasm.gz
    ic-btc-canister.wasm.gz  watchdog-canister.wasm.gz

    # Compute checksums.
    $ sha256sum ic-btc-canister.wasm.gz watchdog-canister.wasm.gz
    09f5647a45ff6d5d05b2b0ed48613fb2365b5fe6573ba0e901509c39fb9564ac  ic-btc-canister.wasm.gz
    cc58b2a32517f9907f0d3c77bc8c099d0a65d8194a8d9bc0ad7df357ee867a07  watchdog-canister.wasm.gz
    ```
4. Attach the Bitcoin Canister's and Watchdog's WASM to the release notes (and nothing else).
    - Clearly state the release status both for Bitcoin mainnet canister and Watchdog mainnet canister in release notes, eg
        - `No change, same as release/<yyyy-mm-dd>`
        - `Waiting for deployment`
        - `Deployed, <NNS proposal link>`, provide the link to an accepted NNS proposal
    - Example
    ```md
    ## Release Status

    - Bitcoin Mainnet Canister: Waiting for deployment
    - Watchdog Mainnet Canister: No change, same as release/<yyyy-mm-dd>
    ```
5. Finalize the release once all the expected canisters were upgraded
    - Update release statuses with corresponding NNS proposal links
    - Uncheck `Set as a pre-release` box to indicate that it's all deployed

```

Reply:
Thanks @maksymar. My impression from the suggestion is that we're making the release process more complex. I'd rather keep things as simple as possible (i.e. let's not worry about updating the release with the current status, and we'll always deploy both the bitcoin canister and the watchdog with every new release, unless the wasm hasn't changed of course).

Reply:
that's perfectly fine with me, those suggestions were optional.

Reply:
I would care more about the total amount of steps, it reads simpler for me when I see 5 steps with a few sub-steps, rather than flat 11 steps.

Reply:
Sounds good. If these are optional, can you go ahead and approve? Feel free to modify the PR to group the steps if you feel like it'd make them more readable.

Reply:
I removed updating release status in the notes.
Also one difference I noticed is the way releases are named, currently it's `release/<yyyy-mm-dd>`, but in your version it was `Release (<yyyy-mm-dd>)`. I left how it is right now, feel free to change.